tests/micropython/ringio.py

```diff
@@ -46,3 +46,33 @@
     micropython.RingIO(None)
 except TypeError as ex:
     print(type(ex))
+
+try:
+    # Buffer may not be empty
+    micropython.RingIO(bytearray(0))
+except ValueError as ex:
+    print(type(ex))
+
+try:
+    # Buffer may not be too small
+    micropython.RingIO(bytearray(1))
+except ValueError as ex:
+    print(type(ex))
+
+try:
+    # Size may not be too small
+    micropython.RingIO(0)
+except ValueError as ex:
+    print(type(ex))
+
+try:
+    # Buffer may not be too big
+    micropython.RingIO(bytearray(65536))
```

Comment:
This is a very large buffer to allocate, and will probably make this test fail on ports with a small heap (or even with a large heap but fragmented).

Probably the easiest thing to do here is just remove this part of the test (unless you can think of a better way to do it?).  The "too large" path is already covered by passing in a large integer.

Reply:
> to save code the `RingIO(65535)` path is also allocating the whole buffer

Ah, right.



> should this be split to a separate test so we can raise SKIP if a bytearray of 65536 can't be allocated?

Yes, it will need to be separated out.  Then skip if any of the `RingIO()` calls raise `MemoryError`.

Reply:
I split those tests to a new "ringio_big.py" test and added tests that the maximum expected size can be constructed.